homeassistant/components/whirlpool/config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Config flow for Whirlpool Sixth Sense integration."""
+import asyncio
+import logging
+
+import aiohttp
+import voluptuous as vol
+from whirlpool.auth import Auth
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN  # pylint: disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema({CONF_USERNAME: str, CONF_PASSWORD: str})
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    auth = Auth(data[CONF_USERNAME], data[CONF_PASSWORD])
+    try:
+        await auth.do_auth()
+    except (asyncio.TimeoutError, aiohttp.ClientConnectionError) as exc:
+        raise CannotConnect from exc
+
+    if not auth.is_access_token_valid():
+        raise InvalidAuth
+
+    return {"title": data[CONF_USERNAME]}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Whirlpool Sixth Sense."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="user", data_schema=STEP_USER_DATA_SCHEMA
+            )
+
+        errors = {}
+
+        try:
+            info = await validate_input(self.hass, user_input)
+        except CannotConnect:
+            errors["base"] = "cannot_connect"
+        except InvalidAuth:
+            errors["base"] = "invalid_auth"
+        except Exception:  # pylint: disable=broad-except
+            _LOGGER.exception("Unexpected exception")
+            errors["base"] = "unknown"
+        else:
+            await self.async_set_unique_id(
+                user_input[CONF_USERNAME], raise_on_progress=False
+            )
```

Comment:
Is there an account ID we an use here instead?

https://developers.home-assistant.io/docs/config_entries_config_flow_handler/#sometimes-acceptable-sources-for-a-unique-id-for-cloud-services

Reply:
The e-mail I know is unique. The AccountID I have no clue about (the API is not documented). I feel safer using the email here